src/iris/api.py

```diff
@@ -1858,6 +1858,26 @@ def on_put(self, req, resp, app_name):
 
         resp.body = '{}'
 
+    def on_delete(self, req, resp, app_name):
+        if not req.context['is_admin']:
+            raise HTTPUnauthorized('Only admins can remove apps', '')
+
+        session = db.Session()
+
+        try:
+            affected = session.execute('''DELETE FROM `application` WHERE `name` = :app_name''', {'app_name': app_name}).rowcount
+            session.commit()
+        except IntegrityError:
+            raise HTTPBadRequest('Cannot remove app. It has likely already in use.', '')
+        finally:
+            session.close()
+
+        if not affected:
+            raise HTTPBadRequest('No rows changed; app name probably already deleted', '')
+
+        resp.status = HTTP_204
```

Comment:
let's keep the status code consistent with the other apis, i.e. return 200 if body is not empty.